src/Services/QueuedJobService.php

```diff
@@ -235,64 +235,30 @@ public function queueJob(QueuedJob $job, $startAfter = null, $userId = null, $qu
-            'Signature' => $signature,
-            'JobStatus' => [
-                QueuedJob::STATUS_NEW,
-                QueuedJob::STATUS_INIT,
-            ],
-        ];
-
-        $existing = QueuedJobDescriptor::get()
-            ->filter($filter)
-            ->first();
-
-        if ($existing && $existing->ID) {
-            return $existing->ID;
-        }
-
-        $jobDescriptor = new QueuedJobDescriptor();
-        $jobDescriptor->JobTitle = $job->getTitle();
-        $jobDescriptor->JobType = $queueName ? $queueName : $job->getJobType();
-        $jobDescriptor->Signature = $signature;
-        $jobDescriptor->Implementation = get_class($job);
-        $jobDescriptor->StartAfter = $startAfter;
-
-        // no user provided - fallback to job user default
-        if ($userId === null && $job instanceof UserContextInterface) {
-            $userId = $job->getRunAsMemberID();
-        }
+        // Create the initial object
+        $jobDescriptor = $this->createJobDescriptor($job, $signature, $startAfter, $userId, $queueName);
 
-        // still no user - fallback to current user
-        if ($userId === null) {
-            if (Security::getCurrentUser() && Security::getCurrentUser()->exists()) {
-                // current user available
-                $runAsID = Security::getCurrentUser()->ID;
-            } else {
-                // current user unavailable
-                $runAsID = 0;
-            }
-        } else {
-            $runAsID = $userId;
+        try {
+            return $this->findOrMakeJobDescriptorFromSignature($signature, $job, $jobDescriptor, $startAfter);
+        } catch (\Throwable $e) {
+            // note that error here may not be an issue as failing to acquire a job lock is a valid state
+            // which happens when other process claimed the job lock first
+            $this->getLogger()->debug(
+                sprintf(
+                    '[%s] - Queued Jobs - Failed to acquire job lock %s %d %s',
+                    DBDatetime::now()->Rfc2822(),
+                    $e->getMessage(),
+                    $signature,
+                    PHP_EOL
+                ),
+                [
+                    'file' => __FILE__,
+                    'line' => __LINE__,
+                ]
+            );
         }
 
-        $jobDescriptor->RunAsID = $runAsID;
-
-        // use this to populate custom data columns before job is queued
-        // note: you can pass arbitrary data to your job and then move it to job descriptor
-        // this is useful if you need some data that needs to be exposed as a separate
-        // DB column as opposed to serialised data
-        $this->extend('updateJobDescriptorBeforeQueued', $jobDescriptor, $job);
-
-        // copy data
-        $this->copyJobToDescriptor($job, $jobDescriptor);
-
-        $jobDescriptor->write();
-
-        $this->startJob($jobDescriptor, $startAfter);
-
-        return $jobDescriptor->ID;
+        return false;
```

Comment:
```suggestion
        return 0;
```
PHPDOC function signature is to return an int